queries.sql

```diff
@@ -7,4 +7,34 @@ select date_of_birth from animals where name in ('Agumon','Pikachu');
 select name, escape_attempts from animals where weight_kg > 10.5;
 select * from animals where neutered is 'true';
 select * from animals where name <> 'Gabumon';
-select * from animals where weight_kg BETWEEN 10.4 and 17.3;
\ No newline at end of file
+select * from animals where weight_kg BETWEEN 10.4 and 17.3;
+
+begin transaction;
+update animals set species = 'unspecified';
+rollback transaction;
+
+Begin work;
+update animals set species = 'digimon' where name like '%mon'
+update animals set species = 'pokemon' where species is null
+commit work;
+
+Begin work;
+Delete from animals;
+rollback work;
+
+Begin transaction;
+Delete from animals where date_of_birth > '2022-01-01'
+savepoint save1;
+update animals set weight_kg = weight_kg * -1;
+rollback to save1;
+update animals set weight_kg = weight_kg * -1 where weight_kg < 0
+commit transaction;
+
+select count(id) from animals;
+select count (*) from animals where escape_attempts = 0;
+select Round(avg(weight_kg),2) from animals;
+select neutered, sum(escape_attempts) as Escape_Attempts from animals group by neutered;
```

Comment:
We need to find whether the neutered or not neutered animals have the most escape attempts. You can either use `avg` or `max` instead of `sum`.